
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>&lt;no title&gt; &#8212; reactome2py 0.0.3 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <span class="target" id="module-reactome2py.content"></span><p>Content Service for Reactome knowledgebase
<a class="reference external" href="https://reactome.org/ContentService/#/">https://reactome.org/ContentService/#/</a></p>
<dl class="function">
<dt id="reactome2py.content.discover">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">discover</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-446203'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.discover" title="Permalink to this definition">¶</a></dt>
<dd><p>The schema.org for an Event in Reactome knowledgebase</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – An event identifier ex. pathway stable identifier (stId) of pathway</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.disease">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">disease</code><span class="sig-paren">(</span><em class="sig-param">doid=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.disease" title="Permalink to this definition">¶</a></dt>
<dd><p>Query list of diseases</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>doid</strong> – returns list of disease DOID if set to true</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The list of diseases or disease DOID if doid param is set to true</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.entities_complex">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">entities_complex</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-5674003'</em>, <em class="sig-param">exclude_structures=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.entities_complex" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of subunits that constitute any given complex.
In case the complex comprises other complexes, this method recursively traverses the content returning each
contained PhysicalEntity. Contained complexes and entity sets can be excluded setting the ‘exclude_structures’
optional parameter to ‘true’</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – the complex for which subunits are requested</p></li>
<li><p><strong>exclude_structures</strong> – specifies whether contained complexes and entity sets are excluded in the response</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list with the entities contained in a given complex</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.entities_complexes">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">entities_complexes</code><span class="sig-paren">(</span><em class="sig-param">id='P00533'</em>, <em class="sig-param">resource='UniPort'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.entities_complexes" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of complexes that contain a given (identifier, resource). The method deconstructs the complexes
into all its participants to do so.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – the identifier for which complexes are requested</p></li>
<li><p><strong>resource</strong> – the resource of the identifier for complexes are requested</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of complexes containing the pair (identifier, resource)</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.entity_other_form">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">entity_other_form</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-199420'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.entity_other_form" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a list containing all other forms of the given PhysicalEntity.
These other forms are PhysicalEntities that share the same ReferenceEntity identifier,
e.g. PTEN H93R[R-HSA-2318524] and PTEN C124R[R-HSA-2317439] are two forms of PTEN.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – DbId or StId of a PhysicalEntity</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.entity_structures">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">entity_structures</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-199420'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.entity_structures" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the list of structures (Complexes and Sets) that include the given entity as their component.
It should be mentioned that the list includes only simplified entries (type, names, ids) and not full information
about each item.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – </p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of larger structures containing the entity</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.event_ancestors">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">event_ancestors</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-5673001'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.event_ancestors" title="Permalink to this definition">¶</a></dt>
<dd><p>The Reactome definition of events includes pathways and reactions.
Although events are organised in a hierarchical structure, a single event can be in more than one location,
i.e. a reaction can take part in different pathways while, in the same way, a sub-pathway can take part
in many pathways. Therefore, this method retrieves a list of all possible paths from the requested event to the
top level pathway(s).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – The event for which the ancestors are requested</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The ancestors of a given event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.event_species">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">event_species</code><span class="sig-paren">(</span><em class="sig-param">species='9606'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.event_species" title="Permalink to this definition">¶</a></dt>
<dd><p>Events (pathways and reactions) in Reactome are organised in a hierarchical structure for every species.
By following all ‘hasEvent’ relationships, this method retrieves the full event hierarchy for any given species.
The result is a list of tree structures, one for each TopLevelPathway. Every event in these trees is represented
by a PathwayBrowserNode. The latter contains the stable identifier, the name, the species, the url, the type, and
the diagram of the particular event.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>species</strong> – Species name (eg: Homo sapiens) or Species taxId (eg: 9606)</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The full event hierarchy for a given species</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.export_diagram">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">export_diagram</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-177929'</em>, <em class="sig-param">ext='png'</em>, <em class="sig-param">quality='5'</em>, <em class="sig-param">flag_interactors=False</em>, <em class="sig-param">title=True</em>, <em class="sig-param">margin='15'</em>, <em class="sig-param">ehld=True</em>, <em class="sig-param">diagram_profile='Modern'</em>, <em class="sig-param">resource='Total'</em>, <em class="sig-param">analysis_profile='Standard'</em>, <em class="sig-param">token=None</em>, <em class="sig-param">flag=None</em>, <em class="sig-param">sel=[]</em>, <em class="sig-param">exp_column=None</em>, <em class="sig-param">file='report'</em>, <em class="sig-param">path=''</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.export_diagram" title="Permalink to this definition">¶</a></dt>
<dd><p>This method accepts identifiers for Event class instances.
When a diagrammed pathway is provided, the diagram is exported to the specified format.
When a subpathway is provided, the diagram for the parent is exported and the events that are part of the subpathways are selected.
When a reaction is provided, the diagram containing the reaction is exported and the reaction is selected.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – Event identifier (it can be a pathway with diagram, a subpathway or a reaction)</p></li>
<li><p><strong>ext</strong> – File extension (defines the image format) available extensions: png, jpg, jpeg, svg, gif</p></li>
<li><p><strong>quality</strong> – Result image quality between [1 - 10]. It defines the quality of the final image (Default 5)</p></li>
<li><p><strong>flag</strong> – Gene name, protein or chemical identifier or Reactome identifier used to flag elements in the diagram</p></li>
<li><p><strong>flag_interactors</strong> – Defines whether to take into account interactors for the flagging default set to False</p></li>
<li><p><strong>sel</strong> – Highlight element(s) selection in the diagram. CSV line. comma seperate python list ex [‘X’, ‘Y’, ‘Z’]</p></li>
<li><p><strong>token</strong> – The analysis token with the results to be overlaid on top of the given diagram</p></li>
<li><p><strong>title</strong> – Sets whether the name of the pathway is shown as title</p></li>
<li><p><strong>margin</strong> – Defines the image margin between [0 - 20] (Default 15)</p></li>
<li><p><strong>ehld</strong> – Defines whether textbook-like illustration are taken into account</p></li>
<li><p><strong>diagram_profile</strong> – Diagram Color Profile: Modern or Standard</p></li>
<li><p><strong>resource</strong> – The analysis resource for which the results will be overlaid on top of the given pathways overview</p></li>
<li><p><strong>exp_column</strong> – Expression column. When the token is associated to an expression analysis, this parameter allows specifying the expression column for the overlay</p></li>
<li><p><strong>analysis_profile</strong> – Analysis Color Profile: Standard, Strosobar, Copper Plus</p></li>
<li><p><strong>file</strong> – Name of file default is ‘report’</p></li>
<li><p><strong>path</strong> – Absolute path to save the file to</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Exports a given pathway diagram to the specified image format (png, jpg, jpeg, svg, gif)</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.export_document">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">export_document</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-177929'</em>, <em class="sig-param">level='1'</em>, <em class="sig-param">diagram_profile='Modern'</em>, <em class="sig-param">resource='Total'</em>, <em class="sig-param">analysis_profile='Standard'</em>, <em class="sig-param">token=None</em>, <em class="sig-param">exp_column=None</em>, <em class="sig-param">file='report'</em>, <em class="sig-param">path=''</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.export_document" title="Permalink to this definition">¶</a></dt>
<dd><p>This method accepts identifiers for Event class instances.
The generated document contains the details for the given event and, optionally, its children (see level parameter). These details include:
- A diagram image
- Summation
- Literature references
- Edit history
- Other details: type, location, compartments, diseases</p>
<p>Documents can also be overlaid with pathway analysis results (given a token)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – Event identifier (it can be a pathway with diagram, a subpathway or a reaction)</p></li>
<li><p><strong>level</strong> – Number of levels to explore down in the pathways hierarchy [0 - 1]</p></li>
<li><p><strong>diagram_profile</strong> – Diagram Color Profile Modern or Standard</p></li>
<li><p><strong>resource</strong> – The analysis resource for which the results will be overlaid on top of the given pathways overview</p></li>
<li><p><strong>analysis_profile</strong> – Analysis Color Profile: Standard, Strosobar, Copper Plus</p></li>
<li><p><strong>token</strong> – The analysis token with the results to be overlaid on top of the given diagram</p></li>
<li><p><strong>exp_column</strong> – Expression column. When the token is associated to an expression analysis, this parameter allows specifying the expression column for the overlay</p></li>
<li><p><strong>file</strong> – Name of file default is ‘report’</p></li>
<li><p><strong>path</strong> – Absolute path to save the file to</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Exports the content of a given event (pathway or reaction) to a PDF document</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.export_event">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">export_event</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-177929'</em>, <em class="sig-param">format='sbgn'</em>, <em class="sig-param">file='report'</em>, <em class="sig-param">path=''</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.export_event" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Exports a given pathway or reaction to the format requested:</dt><dd><ul class="simple">
<li><p>Systems Biology Graphical Notation (SBGN)</p></li>
<li><p>Systems Biology Markup Language (SBML)</p></li>
</ul>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – DbId or StId of the requested pathway or reaction</p></li>
<li><p><strong>format</strong> – sbgn or sbml</p></li>
<li><p><strong>file</strong> – Name of file default is set to report</p></li>
<li><p><strong>path</strong> – Absolute path to save the file</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Exports a given pathway or reaction to SBGN</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.export_fireworks">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">export_fireworks</code><span class="sig-paren">(</span><em class="sig-param">species='9606'</em>, <em class="sig-param">ext='png'</em>, <em class="sig-param">file='report'</em>, <em class="sig-param">path=''</em>, <em class="sig-param">quality='5'</em>, <em class="sig-param">flag=None</em>, <em class="sig-param">flag_interactors=False</em>, <em class="sig-param">sel=[]</em>, <em class="sig-param">title=True</em>, <em class="sig-param">margin='15'</em>, <em class="sig-param">resource='Total'</em>, <em class="sig-param">diagram_profile=''</em>, <em class="sig-param">coverage=False</em>, <em class="sig-param">token=None</em>, <em class="sig-param">exp_column=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.export_fireworks" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>species</strong> – Species identifier (it can be the taxonomy id, species name or dbId)</p></li>
<li><p><strong>ext</strong> – File extension (defines the image format) available extensions: png, jpg, jpeg, svg, gif</p></li>
<li><p><strong>file</strong> – Name of file default is set to report</p></li>
<li><p><strong>path</strong> – Absolute path to save the file</p></li>
<li><p><strong>quality</strong> – Result image quality between [1 - 10]. It defines the quality of the final image (Default 5)</p></li>
<li><p><strong>flag</strong> – Gene name, protein or chemical identifier or Reactome identifier used to flag elements in the diagram</p></li>
<li><p><strong>flag_interactors</strong> – Defines whether to take into account interactors for the flagging</p></li>
<li><p><strong>sel</strong> – Highlight element(s) selection in the diagram. CSV line. comma seperate python list ex [‘X’, ‘Y’, ‘Z’]</p></li>
<li><p><strong>title</strong> – Sets whether the name of the pathway is shown below</p></li>
<li><p><strong>margin</strong> – Defines the image margin between [0 - 20] (Default 15)</p></li>
<li><p><strong>resource</strong> – The analysis resource for which the results will be overlaid on top of the given pathways overview</p></li>
<li><p><strong>diagram_profile</strong> – Diagram Color Profile available in: Copper, Copper plus, Barium lithium, Calcium salts</p></li>
<li><p><strong>coverage</strong> – Set to ‘true’ to overlay analysis coverage values default is set to false</p></li>
<li><p><strong>token</strong> – The analysis token with the results to be overlaid on top of the given pathways overview</p></li>
<li><p><strong>exp_column</strong> – Expression column. When the token is associated to an expression analysis, this parameter allows specifying the expression column for the overlay</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Exports a given pathway overview to the specified image format (png, jpg, jpeg, svg, gif)</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.export_reaction">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">export_reaction</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-6787403'</em>, <em class="sig-param">ext='png'</em>, <em class="sig-param">file='report'</em>, <em class="sig-param">path=''</em>, <em class="sig-param">quality='5'</em>, <em class="sig-param">flag=None</em>, <em class="sig-param">flag_interactors=False</em>, <em class="sig-param">sel=[]</em>, <em class="sig-param">title=True</em>, <em class="sig-param">margin='15'</em>, <em class="sig-param">resource='Total'</em>, <em class="sig-param">diagram_profile=''</em>, <em class="sig-param">coverage=False</em>, <em class="sig-param">token=None</em>, <em class="sig-param">exp_column=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.export_reaction" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – Reaction identifier</p></li>
<li><p><strong>ext</strong> – File extension (defines the image format) available extensions: png, jpg, jpeg, svg, gif</p></li>
<li><p><strong>file</strong> – Name of file default is set to report</p></li>
<li><p><strong>path</strong> – Absolute path to save the file</p></li>
<li><p><strong>quality</strong> – Result image quality between [1 - 10]. It defines the quality of the final image (Default 5)</p></li>
<li><p><strong>flag</strong> – Gene name, protein or chemical identifier or Reactome identifier used to flag elements in the diagram</p></li>
<li><p><strong>flag_interactors</strong> – Defines whether to take into account interactors for the flagging</p></li>
<li><p><strong>sel</strong> – Highlight element(s) selection in the diagram. CSV line. comma seperate python list ex [‘X’, ‘Y’, ‘Z’]</p></li>
<li><p><strong>title</strong> – Sets whether the name of the pathway is shown below</p></li>
<li><p><strong>margin</strong> – Defines the image margin between [0 - 20] (Default 15)</p></li>
<li><p><strong>resource</strong> – The analysis resource for which the results will be overlaid on top of the given pathways overview</p></li>
<li><p><strong>diagram_profile</strong> – Diagram Color Profile available in: Copper, Copper plus, Barium lithium, Calcium salts</p></li>
<li><p><strong>coverage</strong> – Set to ‘true’ to overlay analysis coverage values default is set to false</p></li>
<li><p><strong>token</strong> – The analysis token with the results to be overlaid on top of the given pathways overview</p></li>
<li><p><strong>exp_column</strong> – Expression column. When the token is associated to an expression analysis, this parameter allows specifying the expression column for the overlay</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_acc_pathways">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_acc_pathways</code><span class="sig-paren">(</span><em class="sig-param">acc='Q9BXM7-1'</em>, <em class="sig-param">species='Homo sapiens'</em>, <em class="sig-param">only_diagrammed=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_acc_pathways" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve a list of lower level pathways where the interacting molecules can be found</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>acc</strong> – Accession</p></li>
<li><p><strong>species</strong> – The species name for which the pathways are requested (e.g. ‘Homo sapiens’)</p></li>
<li><p><strong>only_diagrammed</strong> – Specifies whether the pathways has to have an associated diagram or not</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Retrieve a list of lower level pathways where the interacting molecules can be found</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_form">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_form</code><span class="sig-paren">(</span><em class="sig-param">path</em>, <em class="sig-param">name</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_form" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse file and retrieve a summary associated with a token</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path</strong> – Absolute path to file to be read with custom interactor</p></li>
<li><p><strong>name</strong> – Name which identifies the sample</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_psicquic_acc">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_psicquic_acc</code><span class="sig-paren">(</span><em class="sig-param">resource='MINT'</em>, <em class="sig-param">acc='Q13501'</em>, <em class="sig-param">by='details'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_psicquic_acc" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>if by details</dt><dd><ul class="simple">
<li><p>Retrieve clustered interaction, sorted by score, of a given accession by resource</p></li>
</ul>
</dd>
<dt>if by summary</dt><dd><ul class="simple">
<li><p>Retrieve a summary of a given accession by resource</p></li>
</ul>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>resource</strong> – Proteomics standards initiative common query interface (PSICQUIC) Resource</p>
</dd>
</dl>
<p>use interactors_psicquic_resources to retrive all active resources
:param acc: Single Accession
:param by: details or summary (returns counts of details available)
:return:</p>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_psicquic_accs">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_psicquic_accs</code><span class="sig-paren">(</span><em class="sig-param">proteins='EGFR'</em>, <em class="sig-param">resource='MINT'</em>, <em class="sig-param">by='details'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_psicquic_accs" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>if by details</dt><dd><ul class="simple">
<li><p>Retrieve clustered interaction, sorted by score, of a given accession(s) by resource.</p></li>
</ul>
</dd>
<dt>if by summary</dt><dd><ul class="simple">
<li><p>Retrieve a summary of a given accession list by resource.</p></li>
</ul>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>proteins</strong> – Comma seperate list of Accessions in string format ‘a1,a2,a3’</p></li>
<li><p><strong>resource</strong> – Proteomics standards initiative common query interface (PSICQUIC) Resource</p></li>
</ul>
</dd>
</dl>
<p>use interactors_psicquic_resources to retrive all active resources
:return:</p>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_psicquic_resources">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_psicquic_resources</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_psicquic_resources" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve a list of all Psicquic Registries services</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Retrieve a list of all Psicquic Registries services</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_psicquic_url">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_psicquic_url</code><span class="sig-paren">(</span><em class="sig-param">name</em>, <em class="sig-param">psicquic_url</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_psicquic_url" title="Permalink to this definition">¶</a></dt>
<dd><p>Registry custom PSICQUIC resource</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> – Name which identifies the custom psicquic</p></li>
<li><p><strong>psicquic_url</strong> – A URL pointing to the Custom PSICQUIC Resource</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Registry custom PSICQUIC resource</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_static_acc">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_static_acc</code><span class="sig-paren">(</span><em class="sig-param">acc='Q13501'</em>, <em class="sig-param">page='-1'</em>, <em class="sig-param">page_size='-1'</em>, <em class="sig-param">by='details'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_static_acc" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>acc</strong> – Interactor accession (or identifier)</p></li>
<li><p><strong>page</strong> – For paginating the results</p></li>
<li><p><strong>page_size</strong> – Number of results to be retrieved</p></li>
<li><p><strong>by</strong> – details or summary (returns counts of details available)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_static_accs">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_static_accs</code><span class="sig-paren">(</span><em class="sig-param">accs='Q9BXM7-1'</em>, <em class="sig-param">by='details'</em>, <em class="sig-param">page='-1'</em>, <em class="sig-param">page_size='-1'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_static_accs" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>accs</strong> – Comma seperate list of Accessions in string format ‘a1,a2,a3’</p></li>
<li><p><strong>by</strong> – details or summary (returns counts of details available)</p></li>
<li><p><strong>page</strong> – For paginating the results</p></li>
<li><p><strong>page_size</strong> – Number of results to be retrieved</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_upload_content">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_upload_content</code><span class="sig-paren">(</span><em class="sig-param">name</em>, <em class="sig-param">content</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_upload_content" title="Permalink to this definition">¶</a></dt>
<dd><p>Paste file content and get a summary associated with a token</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> – Name which identifies the sample</p></li>
<li><p><strong>content</strong> – Paste custom interactors file content</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Paste file content and get a summary associated with a token</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.interactors_url">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">interactors_url</code><span class="sig-paren">(</span><em class="sig-param">name</em>, <em class="sig-param">interactors_url</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.interactors_url" title="Permalink to this definition">¶</a></dt>
<dd><p>Send file via URL and get a summary associated with a token</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> – Name which identifies the sample</p></li>
<li><p><strong>interactors_url</strong> – A URL pointing to the Interactors file</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.mapping">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">mapping</code><span class="sig-paren">(</span><em class="sig-param">id='PTEN'</em>, <em class="sig-param">resource='UniPort'</em>, <em class="sig-param">species='9606'</em>, <em class="sig-param">by='pathways'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.mapping" title="Permalink to this definition">¶</a></dt>
<dd><dl>
<dt>by pathways:</dt><dd><p>Entities play different roles in reactions, and reactions are events that conform a pathway. This method retrieves
the pathways for which an identifier plays a role within one or more of their events.</p>
<p>return: The lower level pathways where an identifier can be mapped to</p>
</dd>
<dt>by reactions:</dt><dd><p>Entities play different roles in reactions. This method retrieves the reactions for which an identifier plays a role .
return: The reactions where an identifier can be mapped to</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – The identifier to be mapped</p></li>
<li><p><strong>resource</strong> – The resource name for which the identifier is submitted</p></li>
<li><p><strong>species</strong> – Species for which the result is filtered. Accepts taxonomy id, species name and dbId.</p></li>
</ul>
</dd>
</dl>
<p>Important: when identifier points to chemical, this becomes mandatory and if not provided, the default is ‘Homo sapiens’
:param by: pathways or reactions
:return:</p>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.orthology">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">orthology</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-6799198'</em>, <em class="sig-param">species='49633'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.orthology" title="Permalink to this definition">¶</a></dt>
<dd><p>Reactome uses the set of manually curated human reactions to computationally infer reactions in
twenty evolutionarily divergent eukaryotic species for which high-quality whole-genome sequence data
are available, and hence a comprehensive and high-quality set of protein predictions exists.
Thus, this method retrieves the orthology for any given event or entity in the specified species.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – The event for which the orthology is requested</p></li>
<li><p><strong>species</strong> – The species for which the orthology is requested</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The orthology for a given event or entity</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.orthology_events">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">orthology_events</code><span class="sig-paren">(</span><em class="sig-param">ids</em>, <em class="sig-param">species</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.orthology_events" title="Permalink to this definition">¶</a></dt>
<dd><p>Reactome uses the set of manually curated human reactions to computationally infer reactions in
twenty evolutionarily divergent eukaryotic species for which high-quality whole-genome sequence
data are available, and hence a comprehensive and high-quality set of protein predictions exists.
Thus, this method retrieves the orthologies for any given set of events or entities in the specified species.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ids</strong> – The events identifiers for which the orthology is requested</p></li>
<li><p><strong>species</strong> – The species for which the orthology is requested</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.participants">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">participants</code><span class="sig-paren">(</span><em class="sig-param">id='5205685'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.participants" title="Permalink to this definition">¶</a></dt>
<dd><p>Participants contains a PhysicalEntity (dbId, displayName) and a collection of ReferenceEntities (dbId, name, identifier, url)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – DbId or StId of a PhysicalEntity</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of participants for a given event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.participants_physical_entities">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">participants_physical_entities</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-5205685'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.participants_physical_entities" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves all the PhysicalEntities that take part in a given event. It is worth mentioning that
because a pathway can contain smaller pathways (subpathways), the method also recursively retrieves the
PhysicalEntities from every constituent</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – The event for which the participating PhysicalEntities are requested</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of participating PhysicalEntities for a given event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.participants_reference_entities">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">participants_reference_entities</code><span class="sig-paren">(</span><em class="sig-param">id='5205685'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.participants_reference_entities" title="Permalink to this definition">¶</a></dt>
<dd><p>PhysicalEntity instances that represent, e.g., the same chemical in different compartments, or different
post-translationally modified forms of a single protein, share numerous invariant features such as names,
molecular structure and links to external databases like UniProt or ChEBI. To enable storage of this shared
information in a single place, and to create an explicit link among all the variant forms of what can also be
seen as a single chemical entity, Reactome creates instances of the separate ReferenceEntity class.
A ReferenceEntity instance captures the invariant features of a molecule. This method retrieves the
ReferenceEntities of all PhysicalEntities that take part in a given event. It is worth mentioning that
because a pathway can contain smaller pathways (subpathways), this method also recursively retrieves the
ReferenceEntities for all PhysicalEntities in every constituent pathway.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – The event for which the participating ReferenceEntities are requested</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of participating ReferenceEntities for a given event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.pathway_contained_event">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">pathway_contained_event</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-5673001'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.pathway_contained_event" title="Permalink to this definition">¶</a></dt>
<dd><p>Events are the building blocks used in Reactome to represent all biological processes,
and they include pathways and reactions. Typically, an event can contain other events. For example,
a pathway can contain smaller pathways and reactions. This method recursively retrieves all the events
contained in any given event.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – The event for which the contained events are requested</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>All the events contained in the given event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.pathway_contained_event_atttibute">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">pathway_contained_event_atttibute</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-5673001'</em>, <em class="sig-param">attribute='stId'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.pathway_contained_event_atttibute" title="Permalink to this definition">¶</a></dt>
<dd><p>Events are the building blocks used in Reactome to represent all biological processes, and they
include pathways and reactions. Typically, an event can contain other events. For example, a pathway
can contain smaller pathways (subpathways) and reactions. This method recursively retrieves a single attribute for
each of the events contained in the given event.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – The event for which the contained events are requested</p></li>
<li><p><strong>attribute</strong> – Attribute to be filtered</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>python list object of a single property for each event contained in the given event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.pathways_low_diagram">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">pathways_low_diagram</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-199420'</em>, <em class="sig-param">species=None</em>, <em class="sig-param">all_forms=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.pathways_low_diagram" title="Permalink to this definition">¶</a></dt>
<dd><p>This method traverses the event hierarchy and retrieves the list of all lower level pathways that have a
diagram and contain the given PhysicalEntity or Event.</p>
<p>if all_forms is set to tru: it retrieves the given PhysicalEntity in any of its variant forms. These variant forms
include for example different post-translationally modified versions of a single protein, or the same chemical
in different compartments.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – The entity that has to be present in the pathways</p></li>
<li><p><strong>species</strong> – The species for which the pathways are requested. Taxonomy identifier (eg: 9606) or species name (eg: ‘Homo sapiens’)</p></li>
<li><p><strong>all_forms</strong> – If true, it retrieves the given PhysicalEntity in any of its variant forms.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of lower level pathways with diagram containing a given entity or event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.pathways_low_entity">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">pathways_low_entity</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-199420'</em>, <em class="sig-param">species=None</em>, <em class="sig-param">all_forms=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.pathways_low_entity" title="Permalink to this definition">¶</a></dt>
<dd><p>This method traverses the event hierarchy and retrieves the list of all lower level pathways that contain
the given PhysicalEntity or Event.</p>
<p>if all_forms is set to true, it retrieves the list of all lower level pathways that contain the given
PhysicalEntity in any of its variant forms. These variant forms include for example different post-translationally
modified versions of a single protein, or the same chemical in different compartments.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – The entity that has to be present in the pathways</p></li>
<li><p><strong>species</strong> – The species for which the pathways are requested. Taxonomy identifier (eg: 9606) or species name (eg: ‘Homo sapiens’)</p></li>
<li><p><strong>all_forms</strong> – If set to true, it retrieves the list of all lower level pathways that contain the given PhysicalEntity in any of its variant forms.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of lower level pathways containing a given entity or event</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.pathways_top_level">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">pathways_top_level</code><span class="sig-paren">(</span><em class="sig-param">species='9606'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.pathways_top_level" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves the list of top level pathways for the given species</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>species</strong> – Specifies the species by the taxonomy identifier (eg: 9606) or species name (eg: ‘Homo+sapiens’)</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>All Reactome top level pathways</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.person_id">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">person_id</code><span class="sig-paren">(</span><em class="sig-param">id='0000-0001-5807-0069'</em>, <em class="sig-param">by=None</em>, <em class="sig-param">attribute=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.person_id" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>With only id parameter declared,</dt><dd><p>Retrieves a person in Reactome by his/her OrcidId or DbId.</p>
</dd>
<dt>With id and query by declared with pathway or publication</dt><dd><p>Retrieves a list of pathways authored by a given person. OrcidId, DbId or Email can be used to specify the person.
Retrieves a list of publications authored by a given person. OrcidId, DbId or Email can be used to specify the person.</p>
</dd>
<dt>with id and attribute declared</dt><dd><p>Retrieves a specific person’s property by his/her OrcidId or DbId.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – Person identifier - Can be OrcidId or DbId</p></li>
<li><p><strong>by</strong> – if not None, query by authored: pathways or publications</p></li>
<li><p><strong>attribute</strong> – Attribute to be filtered ex. displayName</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.person_name">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">person_name</code><span class="sig-paren">(</span><em class="sig-param">name='Steve Jupe'</em>, <em class="sig-param">exact=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.person_name" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a list of people in Reactome with either their first or last name partly matching the given name (string).
If exact is set to true, retrieves a list of people in Reactome with either their first or last name matching exactly
the given name (string).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> – Person’s first or last name</p></li>
<li><p><strong>exact</strong> – </p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of people with first or last name partly or exactly matching a given name (string)</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.query_id">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">query_id</code><span class="sig-paren">(</span><em class="sig-param">id='R-HSA-60140'</em>, <em class="sig-param">enhanced=False</em>, <em class="sig-param">attribute=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.query_id" title="Permalink to this definition">¶</a></dt>
<dd><p>This method queries for an entry in Reactome knowledgebase based on the given identifier, i.e. stable id or
database id. It is worth mentioning that the retrieved database object has all its properties and direct
relationships (relationships of depth 1) filled.</p>
<p>if enhanced is set to true it also includes any second level relationships regarding regulations and catalysts.</p>
<p>if attribute name is present it queries the property of the attribute requested</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> – DbId or StId of the requested database object</p></li>
<li><p><strong>enhanced</strong> – boolean value to make an enhanced query on id</p></li>
<li><p><strong>attribute</strong> – Attribute to be queried</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.query_ids">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">query_ids</code><span class="sig-paren">(</span><em class="sig-param">ids='R-HSA-60140'</em>, <em class="sig-param">mapping=False</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.query_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>This method queries for a set of entries in Reactome knowledgebase based on the given list of identifiers.
The provided list of identifiers can include stable ids, database ids or a mixture of both. It should be
underlined that any duplicated ids are eliminated while only requests containing up to 20 ids are processed.</p>
<p>if mapping is set, to true previous version of stable identifiers can be queried.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ids</strong> – A comma separated list of identifiers</p></li>
<li><p><strong>mapping</strong> – If set to true, retrieves a list of entries with their mapping to the provided identifiers</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.references">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">references</code><span class="sig-paren">(</span><em class="sig-param">id='15377'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.references" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a list containing all the reference entities for a given identifier.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>id</strong> – Identifier for a given entity</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>All ReferenceEntities for a given identifier</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.schema">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">schema</code><span class="sig-paren">(</span><em class="sig-param">name='Pathway'</em>, <em class="sig-param">by=None</em>, <em class="sig-param">species=None</em>, <em class="sig-param">page=None</em>, <em class="sig-param">offset=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.schema" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves the list of entries in Reactome that belong to the specified schema class.
Please take into account that if species is specified to filter the results, schema class needs to be an
instance of Event or PhysicalEntity. Additionally, paging is required, while a maximum of 25 entries can be
returned per request.</p>
<dl class="simple">
<dt>if by is count:</dt><dd><p>counts the total number of entries in Reactome that belong to the specified schema class.</p>
</dd>
<dt>if by is min:</dt><dd><p>the list of simplified entries in Reactome that belong to the specified schema class. A simplified entry may
be considered as a minimised version of the full database object that includes its database id, stable id,
displayName and type.</p>
</dd>
<dt>if by is reference:</dt><dd><p>the list of simplified reference objects that belong to the specified schema class. A reference object
includes its database id, external identifier, and external database name.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>name</strong> – Schema class name.</p></li>
<li><p><strong>by</strong> – if not None, by count, min, or reference (name needs to an instance of ReferenceEntity or ExternalOntology)</p></li>
<li><p><strong>species</strong> – Allowed species filter: SpeciesName (eg: Homo sapiens) SpeciesTaxId (eg: 9606)</p></li>
<li><p><strong>offset</strong> – Number of rows returned. Maximum = 25</p></li>
<li><p><strong>page</strong> – Page to be returned</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_diagram">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_diagram</code><span class="sig-paren">(</span><em class="sig-param">diagram='R-HSA-8848021'</em>, <em class="sig-param">query='MAD'</em>, <em class="sig-param">types=[]</em>, <em class="sig-param">start=None</em>, <em class="sig-param">rows=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_diagram" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs a Apache Solr query (diagram widget scoped) for a given QueryObject</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>diagram</strong> – diagram/pathway stable id</p></li>
<li><p><strong>query</strong> – query names by this string</p></li>
<li><p><strong>types</strong> – types to filter</p></li>
<li><p><strong>start</strong> – start row</p></li>
<li><p><strong>rows</strong> – number of rows to include</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_diagram_instance">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_diagram_instance</code><span class="sig-paren">(</span><em class="sig-param">diagram='R-HSA-68886'</em>, <em class="sig-param">instance='R-HSA-141433'</em>, <em class="sig-param">types=[]</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_diagram_instance" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs a Apache Solr query (diagram widget scoped) for a given QueryObject</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>diagram</strong> – diagram/pathway stable id</p></li>
<li><p><strong>instance</strong> – instance stable id</p></li>
<li><p><strong>types</strong> – types to filter</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_diagram_pathway_flag">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_diagram_pathway_flag</code><span class="sig-paren">(</span><em class="sig-param">diagram='R-HSA-446203'</em>, <em class="sig-param">query='CTSA'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_diagram_pathway_flag" title="Permalink to this definition">¶</a></dt>
<dd><p>This method traverses the content and checks not only for the main identifier but also for all the cross-references to find the flag targets</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>diagram</strong> – diagram/pathway stable id</p></li>
<li><p><strong>query</strong> – query names by this string</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of diagram entities plus pathways from the provided list containing the specified identifier</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_facet">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_facet</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_facet" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves faceting information on the whole Reactome search data.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of facets corresponding to the whole Reactome search data</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_facet_query">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_facet_query</code><span class="sig-paren">(</span><em class="sig-param">query='TP53'</em>, <em class="sig-param">species=[]</em>, <em class="sig-param">types=[]</em>, <em class="sig-param">compartments=[]</em>, <em class="sig-param">keywords=[]</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_facet_query" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves faceting information on a specific query</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>query</strong> – search term</p></li>
<li><p><strong>species</strong> – Species identifier (it can be the taxonomy id, species name or dbId) - python list of strings</p></li>
<li><p><strong>types</strong> – types to filter by - python list of strings</p></li>
<li><p><strong>compartments</strong> – compartments - python list of strings</p></li>
<li><p><strong>keywords</strong> – keywords - python list of strings</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_fireworks">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_fireworks</code><span class="sig-paren">(</span><em class="sig-param">query='BRAF'</em>, <em class="sig-param">species='Homo sapiens'</em>, <em class="sig-param">types=[]</em>, <em class="sig-param">start=None</em>, <em class="sig-param">rows=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_fireworks" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs a Apache Solr query (fireworks widget scoped) for a given QueryObject</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>query</strong> – Search term</p></li>
<li><p><strong>species</strong> – Species identifier (it can be the taxonomy id, species name or dbId)</p></li>
<li><p><strong>types</strong> – Types to filter by - python list of strings</p></li>
<li><p><strong>start</strong> – Start row</p></li>
<li><p><strong>rows</strong> – Number of rows to include</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_fireworks_flag">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_fireworks_flag</code><span class="sig-paren">(</span><em class="sig-param">query='KNTC1'</em>, <em class="sig-param">species='Homo sapiens'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_fireworks_flag" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs a Apache Solr query (fireworks widget scoped) for a given QueryObject</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>query</strong> – Search term</p></li>
<li><p><strong>species</strong> – Species identifier (it can be the taxonomy id, species name or dbId)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_query">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_query</code><span class="sig-paren">(</span><em class="sig-param">query='Biological oxidations'</em>, <em class="sig-param">species=[]</em>, <em class="sig-param">types=[]</em>, <em class="sig-param">compartments=[]</em>, <em class="sig-param">keywords=[]</em>, <em class="sig-param">cluster=True</em>, <em class="sig-param">start=None</em>, <em class="sig-param">rows=None</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_query" title="Permalink to this definition">¶</a></dt>
<dd><p>This method performs a Solr query on the Reactome knowledgebase. Results can be provided in a paginated format.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>query</strong> – term to search</p></li>
<li><p><strong>species</strong> – list of species - python list of strings</p></li>
<li><p><strong>types</strong> – Types to filter by - python list of strings</p></li>
<li><p><strong>compartments</strong> – Compartments - python list of strings</p></li>
<li><p><strong>keywords</strong> – Keywords - python list of strings</p></li>
<li><p><strong>cluster</strong> – Cluster results</p></li>
<li><p><strong>start</strong> – Start row</p></li>
<li><p><strong>rows</strong> – number of rows to include</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_spellcheck">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_spellcheck</code><span class="sig-paren">(</span><em class="sig-param">query='repoduction'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_spellcheck" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves a list of spell-check suggestions for a given search term.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>query</strong> – term to search</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>python list of matched elements</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.search_suggest">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">search_suggest</code><span class="sig-paren">(</span><em class="sig-param">query='platele'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.search_suggest" title="Permalink to this definition">¶</a></dt>
<dd><p>This method retrieves a list of suggestions for a given search term.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>query</strong> – term to search</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>python list of matched elements</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.species">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">species</code><span class="sig-paren">(</span><em class="sig-param">by='all'</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.species" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Query species by:</dt><dd><p>all, returns list of all species in Reactome
main, return list of main species in Reactome</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>by</strong> – all or main</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="reactome2py.content.token_interactors">
<code class="sig-prename descclassname">reactome2py.content.</code><code class="sig-name descname">token_interactors</code><span class="sig-paren">(</span><em class="sig-param">token</em>, <em class="sig-param">proteins</em><span class="sig-paren">)</span><a class="headerlink" href="#reactome2py.content.token_interactors" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve custom interactions associated with a token</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>token</strong> – A token associated with a data submission</p></li>
<li><p><strong>proteins</strong> – Interactors accessions</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Retrieve custom interactions associated with a token</p>
</dd>
</dl>
</dd></dl>



          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">reactome2py</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Nasim Sanati.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 2.2.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/content.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>